lib/order.rb

```diff
@@ -0,0 +1,60 @@
+require_relative 'menu'
+require_relative 'basket'
+
+class Order
+  attr_reader :basket, :menu
+  def initialize
+    @basket = Basket.new
+    @menu = Menu.new
+  end
+  def add(item, qty)
+    raise not_on_menu_error unless menu.has?(item)
+    basket.add(item, qty)
+  end
+  def remove(item, qty)
+    raise not_in_basket_error unless basket.has?(item)
+    basket.remove(item, qty)
+  end
+  def cancel
+    @basket  = Basket.new
+  end
+  def buy
+    print "#{confirmation_msg} #{delivery_time}\n\n"
+  end
+  def view
+    display_header
+    display_basket
+    display_total
+  end
+
+  private
+  def not_on_menu_error
+    "The requested item is not on the menu"
+  end
+  def not_in_basket_error
+    "You have not ordered this item"
+  end
+  def confirmation_msg
+    "Thank you. Your order was placed and will be delivered before"
+  end
+  def delivery_time
+    "#{(Time.now.hour + 1).modulo(24)}:#{Time.now.min}"
```

Comment:
Nice — good catch on the modulo thing. This would have been a good opportunity to test `#buy` to make sure it behaves well at 11pm, midnight, 1am etc — and thereby a good opportunity to use mocking to ensure you didn't have to stay up until midnight just to run your tests.